src/Shared/UrlDecoder/UrlDecoder.cs

```diff
@@ -26,7 +26,7 @@ public static int DecodeRequestLine(ReadOnlySpan<byte> source, Span<byte> destin
 
             // This requires the destination span to be larger or equal to source span
             source.CopyTo(destination);
-            return DecodeInPlace(destination, isFormEncoding);
+            return DecodeInPlace(destination.Slice(0, source.Length), isFormEncoding);
```

Comment:
I added Slice here (similarly to char) because otherwise the returned length was incorrect for `destination`-s larger than the input.